pkg/client/kubernetes/cache/cache.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"strings"
+
+	"github.com/gardener/gardener/pkg/logger"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	runtimecache "sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ cache.Cache = &Aggregator{}
+
+// Aggregator is a cache that can hold different cache implementations depending on the objects' GVKs.
+type Aggregator struct {
+	fallback   cache.Cache
+	gvkToCache map[schema.GroupVersionKind]cache.Cache
+	scheme     *runtime.Scheme
+}
+
+func (c *Aggregator) cacheForObject(obj runtime.Object) cache.Cache {
```

Comment:
Should `client.Object` be used here?

Reply:
Unfortunately, this doesn't work for the `List` function because there we have `client.ObjectList` which is not compatible with `client.Client`. So I chose `runtime.Object` wich works for `client.Object` (from `func Get`) and `client.ObjectList` (from `func Get`)

Reply:
OK, makes sense, thanks!